sync/landing.py

```diff
@@ -233,24 +233,15 @@ def add_pr(self, pr_id, sync, wpt_commits, copy=True, prev_wpt_head=None):
 
         logger.info("Upstream files changed:\n%s" % "\n".join(sorted(upstream_changed)))
 
-        if isinstance(sync, upstream.UpstreamSync):
-            remote_paths = list(upstream_changed)
-            local_paths = [os.path.join(env.config["gecko"]["path"]["wpt"], path)
-                           for path in upstream_changed]
-            has_changes = False
-            for local_path, remote_path in zip(local_paths, remote_paths):
-                try:
-                    if (self.git_wpt.git.show("origin/master:%s" % remote_path) !=
-                        git_work_gecko.git.show("HEAD:%s" % local_path)):
-                        logger.info("Changes in %s" % local_path)
-                        has_changes = True
-                        break
-                except git.GitCommandError:
-                    has_changes = True
-                    break
-            if not has_changes:
-                logger.info("Upstream sync doesn't introduce any gecko changes")
-                return
+        # If this is originally an UpstreamSync and no new changes were introduced to the GH PR
+        # then we can safely skip and not need to re-apply these changes. Compare the hash of
+        # the upstreamed gecko commits against the final hash in the PR.
+        pr_ref = 'origin/pr/{}'.format(pr_id)
+        if (isinstance(sync, upstream.UpstreamSync) and
+                sync.wpt_commits.head.sha1 == sync.git_wpt.refs[pr_ref].commit.hexsha):
```

Comment:
I think this is logically correct, but can we create a `pr_commits` property on  `UpstreamSync` that returns thse commits and handles cases like the sync not having a PR? 

Also the indentation of the second line looks incorrect.

Reply:
That sounds like a good idea, I will work on implementating that.

Its done to avoid it being confused as the next logical line. But obviously can move it back an indent to stick with the styling of the rest of the codebase.